front/lib/api/spaces.ts

```diff
@@ -110,6 +112,20 @@ export async function softDeleteSpaceAndLaunchScrubWorkflow(
       }
     }
 
+    if (options?.force) {
+      const agentNames = uniq(usages.map((u) => u.agentNames).flat());
+      for (const agentName of agentNames) {
+        const res = await updateAgentRequestedGroupIds(
+          auth,
+          { agentName: agentName, newGroupIds: [] },
```

Comment:
This is inaccurate they might have other requestedGroupIds. You want to remove a specific groupId.

What is the code that computes requestedGroupIds on agentConfiguration save. Let's reuse it asking it to skip that space to maximize code reuse

Reply:
(Note here that this code in its inacurracy would create a pretty serious cross restricted space leakage vector)